com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/DetectAppsToRenameStep.java

```diff
@@ -0,0 +1,140 @@
+package com.sap.cloud.lm.sl.cf.process.steps;
+
+import com.sap.cloud.lm.sl.cf.core.model.ApplicationUpdateState;
+import com.sap.cloud.lm.sl.cf.core.model.DeployedMta;
+import com.sap.cloud.lm.sl.cf.core.model.DeployedMtaModule;
+import com.sap.cloud.lm.sl.cf.process.Constants;
+import com.sap.cloud.lm.sl.cf.process.message.Messages;
+import org.cloudfoundry.client.lib.domain.CloudApplication;
+import org.cloudfoundry.client.lib.domain.ImmutableCloudApplication;
+import org.flowable.engine.delegate.DelegateExecution;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.context.annotation.Scope;
+
+import javax.inject.Named;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Set;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Named("detectAppsToRenameStep")
+@Scope(BeanDefinition.SCOPE_PROTOTYPE)
+public class DetectAppsToRenameStep extends SyncFlowableStep {
+
+    private static final String OLD_SUFFIX = ApplicationUpdateState.OLD.asSuffix();
+    private static final String NEW_SUFFIX = ApplicationUpdateState.NEW.asSuffix();
+
+    @Override
+    protected StepPhase executeStep(ExecutionWrapper execution) {
+        StepsUtil.setAppsToUndeploy(execution.getContext(), Collections.emptyList());
+        if (!shouldExecuteStep(execution.getContext())) {
+            return StepPhase.DONE;
+        }
+        DeployedMta deployedMta = StepsUtil.getDeployedMta(execution.getContext());
+        if (deployedMta == null) {
+            return StepPhase.DONE;
+        }
+
+        Set<String> appsToProcess = deployedMta.getModules()
+                                               .stream()
+                                               .map(DeployedMtaModule::getAppName)
+                                               .collect(Collectors.toSet());
+        List<String> appsToRename = new LinkedList<>();
+        List<String> appsToUndeploy = new LinkedList<>();
+
+        for (String appName : appsToProcess) {
```

Comment:
Extract a new method that processes the applications.

Reply:
Done